pkg/noderesourcetopology/filter.go

```diff
@@ -68,28 +68,51 @@ func resourcesAvailableInAnyNUMANodes(logKey string, numaNodes NUMANodeList, res
 
 	// Node() != nil already verified in Filter(), which is the only public entry point
 	nodeName := nodeInfo.Node().Name
+	nodeResources := util.ResourceList(nodeInfo.Allocatable)
 
 	for resource, quantity := range resources {
+		if quantity.IsZero() {
+			// why bother? everything's fine from the perspective of this resource
+			klog.V(4).InfoS("ignoring zero-qty resource request", "logKey", logKey, "node", nodeName, "resource", resource)
+			continue
+		}
+
+		// do we have this resource at node level, and if so, do we have enough to met the requirement?
+		isNodeResource := false
+		if nodeQuantity, ok := nodeResources[resource]; ok && nodeQuantity.Cmp(quantity) >= 0 {
```

Comment:
If ```!ok``` i.e. ```isNodeResource == false``` we should bailout, because if we can't find it at the node level, then a fortiori we can't find it at the NUMA level.

Reply:
we can have resources only at node level though, like extended resources (https://github.com/kubernetes-sigs/scheduler-plugins/issues/355) or device plugins which don't report NUMA affinity

Reply:
I know that it might not be at NUMA level, only at the node level.
But What I'm saying is, if you don't even have it at the **node** level, you can call ```continue``` here.

Reply:
right, I misread initially. That's true indeed, we can simplify the flow here.